File: Atcoder_#331/ruby/b_vuy_one_carton_of_milk.rb
n, s, m, l = gets.split.map(&:to_i)

min_cost = Float::INFINITY

(0..(n/6 + 1)).each do |i|
  (0..(n/8 + 1)).each do |j|
    (0..(n/12 + 1)).each do |k|
      total_eggs = i * 6 + j * 8 + k * 12
      if total_eggs >= n
        cost = i * s + j * m + k * l
        min_cost = [min_cost, cost].min
      end
    end
  end
end

puts min_cost

File: Atcoder_#331/javascript/b_vuy_one_carton_of_milk.js
function main(input) {
  const [n, s, m, l] = input.trim().split(' ').map(Number);

  let minCost = Infinity;

  for (let i = 0; i <= Math.floor(n / 6) + 1; i++) {
      for (let j = 0; j <= Math.floor(n / 8) + 1; j++) {
          for (let k = 0; k <= Math.floor(n / 12) + 1; k++) {
              const totalEggs = i * 6 + j * 8 + k * 12;
              if (totalEggs >= n) {
                  const cost = i * s + j * m + k * l;
                  minCost = Math.min(minCost, cost);
              }
          }
      }
  }

  console.log(minCost);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));
